### PR TITLE
gateway: Fix Sec-WebSocket-Key accept handling (#8166)

### PR DESCRIPTION
**Pull request by mfleisz**
Fixes the issue reported in #8166

### CODE REVIEW

File: libfreerdp/core/gateway/http.c [hunk 1]
```diff
@@ -1128,7 +1127,7 @@ BOOL http_response_is_websocket(HttpContext* http, HttpResponse* response)
 	if (!winpr_Digest_Init(sha1, WINPR_MD_SHA1))
 		goto out;
 
-	if (!winpr_Digest_Update(sha1, http->SecWebsocketKey, sizeof(http->SecWebsocketKey)))
+	if (!winpr_Digest_Update(sha1, http->SecWebsocketKey, strlen(http->SecWebsocketKey)))
```
Comment (akarl10) [inline libfreerdp/core/gateway/http.c:1130]: you already fixed that previously by making http->SecWebsocketKey of type BYTE, but now gcc complains about the second argument being a char*. I suggest casting that to BYTE. strlen should be correct.

**Comment by akarl10** [comment]
this patch works for me.

File: libfreerdp/core/gateway/http.c (hunk 1 above)
Comment (mfleisz) [reply libfreerdp/core/gateway/http.c]: You'r right - thx for pointing out

**Comment by akarl10** [comment]
LGTM

**Review by akallabeth — APPROVED** [review]

**Pull request merged**